### PR TITLE
Unhardcode leaky_bionic

### PR DESCRIPTION
**Pull request by Ramza13**
#### Summary
None

#### Purpose of change
Unhardcode leaky_bionic, add more example effect_on_conditions, more dialog functions

#### Describe the solution
Added dialog function to mod health and move leaky_bionic to use it.  Also changed add_power to use talker more rather than sidestepping it.

### CODE REVIEW

**Comment by actual-nh** [comment]
Something to adjust morale should enable doing a better fix (than #47894) for Alonso, BTW.

**Comment by Ramza13** [comment]
> 
> 
> Something to adjust morale should enable doing a better fix (than #47894) for Alonso, BTW.

I already have the functions to add/remove morale written as part of #42786 only reason I haven't used them was lack of a use case.  Thats not a bad one though I don't relish the idea of deciding numbers for that particular buff.  If you have numbers/name of the morale effect you want to see, I can make that PR asap as opposed to whenever I can work up the motivation to quantify sex.

**Comment by actual-nh** [comment]
> > Something to adjust morale should enable doing a better fix (than #47894) for Alonso, BTW.
> 
> I already have the functions to add/remove morale written as part of #42786 only reason I haven't used them was lack of a use case. Thats not a bad one though I don't relish the idea of deciding numbers for that particular buff. If you have numbers/name of the morale effect you want to see, I can make that PR asap as opposed to whenever I can work up the motivation to quantify sex.

@souricelle, any ideas? I'm thinking at minimum enough to overcome traits like Bad Temper's -10 (even with its reduction of positive morale effects).

**Comment by LyleSY** [comment]
RE: Alonso's effect, I was just trying to copy the vibrator morale effect in a hacky way. I could see an argument for it being twice as good.

File: data/json/effect_on_condition.json [hunk 1]
```diff
@@ -151,5 +151,14 @@
       ]
     },
     "effect": [ { "message": "NIGHT_MESSAGES_3", "snippet": true } ]
+  },
+  {
+  "type": "effect_on_condition",
```
Comment (kevingranade) [inline data/json/effect_on_condition.json:156]:
```suggestion
    "type": "effect_on_condition",
```

File: doc/NPCs.md [hunk 2]
```diff
@@ -492,6 +492,7 @@ Effect | Description
 `u_mod_fatigue: fatigue_int`<br/>`npc_mod_fatigue: fatigue_int` | Your character or the NPC will have `fatigue_int` added or subtracted from its fatigue.
 `u_make_sound, npc_make_sound: message_string`, `volume: volume_int`, `type: type_string`,  | A sound of description `message_string` will be made at your character or the NPC's location of volume `volume_int` and type `type_string`. Possible types are: background, weather, music, movement, speech, electronic_speech, activity, destructive_activity, alarm, combat,    alert, or order
 
+`u_mod_healthy, npc_mod_healthy : amount_int, cap: cap_int`<br/> | Your character or the NPC will have `amount_int` added or subtracted from its health value to a cap of `cap_int`.
```
Comment (ZhilkinSerg) [inline doc/NPCs.md:495]:
```suggestion
`u_mod_healthy, npc_mod_healthy : amount_int, cap: cap_int` | Your character or the NPC will have `amount_int` added or subtracted from its health value to a cap of `cap_int`.
```

File: doc/NPCs.md [hunk 3]
```diff
@@ -492,6 +492,7 @@ Effect | Description
 `u_mod_fatigue: fatigue_int`<br/>`npc_mod_fatigue: fatigue_int` | Your character or the NPC will have `fatigue_int` added or subtracted from its fatigue.
 `u_make_sound, npc_make_sound: message_string`, `volume: volume_int`, `type: type_string`,  | A sound of description `message_string` will be made at your character or the NPC's location of volume `volume_int` and type `type_string`. Possible types are: background, weather, music, movement, speech, electronic_speech, activity, destructive_activity, alarm, combat,    alert, or order
 
+`u_mod_healthy, npc_mod_healthy : amount_int, cap: cap_int` | Your character or the NPC will have `amount_int` added or subtracted from its health value to a cap of `cap_int`.
 #### Trade / Items
```
Comment (actual-nh) [inline doc/NPCs.md:496]:
```suggestion
`u_mod_healthy, npc_mod_healthy : amount_int, cap: cap_int` | Your character or the NPC will have `amount_int` added or subtracted from its health value, but not beyond `cap_int`.

#### Trade / Items
```
Exactly how does the cap work?

**Pull request merged**